xclim/indices/generic.py

```diff
@@ -147,97 +162,205 @@ def fitfunc(arr):
     out.attrs["estimator"] = "Maximum likelihood"
     out.attrs["scipy_dist"] = dist
     out.attrs["units"] = ""
-    # out.name = 'params'
+
     return out
 
 
-def fa(
-    da: xr.DataArray, t: Union[int, Sequence], dist: str = "norm", mode: str = "high"
-):
-    """Return the value corresponding to the given return period.
+def pwm_fit(da: xr.DataArray, dist: str = "norm"):
+    """Fit an array to a univariate distribution along the time dimension using Probability Weighted Moments.
 
     Parameters
     ----------
     da : xr.DataArray
-      Maximized/minimized input data with a `time` dimension.
-    t : Union[int, Sequence]
-      Return period. The period depends on the resolution of the input data. If the input array's resolution is
-      yearly, then the return period is in years.
-    dist : str
-      Name of the univariate distribution, such as beta, expon, genextreme, gamma, gumbel_r, lognorm, norm
-      (see scipy.stats).
-    mode : {'min', 'max}
-      Whether we are looking for a probability of exceedance (max) or a probability of non-exceedance (min).
+      Time series to be fitted along the time dimension.
+
+    Returns
+    -------
+    xr.DataArray
+      An array of distribution parameters fitted using the method of Maximum Likelihood.
+    dist : {'expon', 'gamma', 'genextreme', 'genpareto', 'gumbel_r', 'pearson3', 'weibull_min'}
+      Name of the univariate distribution. Subset of scipy distributions supported by the `lmoments3` package.
+
+    Notes
+    -----
+    Probability Weighted Moments are also known as L-moments. This function uses an implementation from the
+    `lmoments3` package, which is not actively maintained. Ideally, parameter estimation from probability
+    weighted moments would be implemented directly into Scipy.
+
+    Coordinates for which all values are NaNs will be dropped before fitting the distribution. If the array
+    still contains NaNs, the distribution parameters will be returned as NaNs.
+    """
+    import dask
+
+    dc = get_dist(dist)
+    lm3dc = get_lm3_dist(dist)
+
+    shape_params = [] if dc.shapes is None else dc.shapes.split(",")
+    dist_params = shape_params + ["loc", "scale"]
+
+    # Fit the parameters.
+    # This would also be the place to impose constraints on the series minimum length if needed.
+    def fitfunc(arr):
+        """Fit distribution parameters."""
+        x = np.ma.masked_invalid(arr).compressed()
+
+        # Return NaNs if array is empty.
+        if len(x) <= 1:
+            return [np.nan] * len(dist_params)
+
+        # Fill with NaNs if one of the parameters is NaN
+        params = list(lm3dc.lmom_fit(x).values())
+        if np.isnan(params).any():
+            params[:] = np.nan
+
+        return params
+
+    # xarray.apply_ufunc does not yet support multiple outputs with dask parallelism.
+    data = dask.array.apply_along_axis(fitfunc, da.get_axis_num("time"), da)
```

Comment:
I'd suggest making this a duck ops : dask methods will return dask arrays, which is not always wanted!
A simple way, for this specific case is :
`duck = dask.array if isintance(da.data, dask.array.Array) else numpy`
since `np.apply_along_axis` has the same signature as the dask version.